frontend/ui/elements/common_settings_menu_table.lua

```diff
@@ -615,7 +642,19 @@ common_settings.document_save = {
     text = _("Save document (write highlights into PDF)"),
     sub_item_table = {
         genGenericMenuEntry(_("Prompt"), "save_document", "prompt", "prompt"), -- set "save_document" to "prompt"
-        genGenericMenuEntry(_("Always"), "save_document", "always"),
+        {
+            text = _("Always"),
+            checked_func = function()
+                return G_reader_settings:readSetting("save_document") == "always"
+            end,
+            callback = function()
+                logger.warn("testing!")
+                if G_reader_settings:readSetting("document_metadata_folder") == "hash" then
+                    UIManager:show(InfoMessage:new{ text = "Warning: Book metadata location is set to hash-based storage. Writing highlights into a PDF modifies the file which may change the partial hash, resulting in its metadata (e.g., highlights and progress) being unlinked and lost.", })
```

Comment:
The text should be translatable (`_("text")`).

Reply:
I'm guessing this somehow links this text back to the i10n module to make it possible to translate (even if the string later changes), that's all that's needed here, rigiht? Since I only speak English and Spanish..

Reply:
Yes, the string is the key and if it's not found it falls back to what was put in.